staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler_test.go

```diff
@@ -277,3 +276,33 @@ func TestGetObjectMetaNils(t *testing.T) {
 		t.Errorf("expected labels to be %v, got %v", expected, got)
 	}
 }
+
+func TestGetObjectMeta(t *testing.T) {
+	for i := 0; i < 100; i++ {
+		u := &unstructured.Unstructured{Object: map[string]interface{}{
+			"metadata": map[string]interface{}{
+				"name": "good",
+				"Name": "bad1",
+				"nAme": "bad2",
+				"naMe": "bad3",
+				"namE": "bad4",
+
+				"namespace": "good",
+				"Namespace": "bad1",
+				"nAmespace": "bad2",
+				"naMespace": "bad3",
+				"namEspace": "bad4",
+
+				"creationTimestamp": "a",
+			},
+		}}
```

Comment:
remembering discussion about deeply nested fields in the json iterator patch thread, behaving differently than toplevel fields. Maybe add a `mEtaData`?

Reply:
> remembering discussion about deeply nested fields in the json iterator patch thread, behaving differently than toplevel fields. Maybe add a mEtaData?

that ended up being due to the way json-iter treats structs with < 10 fields differently, not a nesting issue. I think that is covered by the json-iter unit tests